plugins/modules/keycloak_userprofile.py

```diff
@@ -0,0 +1,727 @@
+
+                            required:
+                                description:
+                                    - The roles that require this attribute.
+                                type: dict
+                                suboptions:
+                                    roles:
+                                        description:
+                                            - The roles for which this attribute is required.
+                                            - Supported values are V(admin) and V(user).
+                                        type: list
+                                        elements: str
+                                        default:
+                                          - user
+
+                    groups:
+                        description:
+                            - A list of attribute groups to be included in the User Profile.
+                        type: list
+                        elements: dict
+                        suboptions:
+                            name:
+                                description:
+                                    - The name of the group.
+                                type: str
+                                required: true
+
+                            display_header:
+                                description:
+                                    - The display header for the group.
+                                aliases:
+                                  - displayHeader
+                                type: str
+                                required: true
+
+                            display_description:
+                                description:
+                                    - The display description for the group.
+                                aliases:
+                                  - displayDescription
+                                type: str
+                                required: false
+
+                            annotations:
+                                description:
+                                    - The annotations included in the group.
+                                type: dict
+                                required: false
+
+                    unmanaged_attribute_policy:
+                        description:
+                            - Policy for unmanaged attributes.
+                            - Supported values are V(ENABLED), V(ADMIN_EDIT) and V(ADMIN_VIEW).
+                        aliases:
+                          - unmanagedAttributePolicy
+                        type: str
+                        choices:
+                            - ENABLED
+                            - ADMIN_EDIT
+                            - ADMIN_VIEW
+
+notes:
+    - Currently, only a single V(declarative-user-profile) entry is supported for O(provider_id) (design of the Keyckoak API).
+    - However, there can be multiple O(config.kc_user_profile_config[].attributes[]) entries.
```

Comment:
I think this should be part of the previous note.
```suggestion
      However, there can be multiple O(config.kc_user_profile_config[].attributes[]) entries.
```

Reply:
:+1: